### PR TITLE
test: Update check-openshift after kubevirt 0.2.0 release

### PR DESCRIPTION
**Pull request by mareklibra**
With new Kubevirt 0.2.0 release, the `virt-api` pod is no more present.
The health check can be performed on `virt-controler` pod instead.

 * [x] image-refresh openshift

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-rc709.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180112-091010/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180112-091010/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-1dl5p.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180112-095757/

**Comment by cockpituous** [comment]
Task aborted.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-h0jkf.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180112-123030/

**Comment by cockpituous** [comment]
image-refresh openshift done: https://github.com/cockpituous/cockpit/commits/image-refresh-openshift-20180112-130707

**Comment by mareklibra** [comment]
`docker.io/registry` (image ID `d1fd7d86a825`) was properly tagged to `:2` after `docker pull` but seems to loose its tag to `<none>` on its way towards end of `openshift.setup` script. Needs farther investigation.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-qtzg9.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180115-095656/

**Comment by cockpituous** [comment]
Task aborted.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-0rt9w.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180115-123838/

**Comment by cockpituous** [comment]
Task aborted.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-17wnw.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180115-151414/

**Comment by cockpituous** [comment]
Task aborted.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-bjb38.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180115-171313/

**Comment by cockpituous** [comment]
image-refresh openshift done: https://github.com/cockpituous/cockpit/commits/image-refresh-openshift-20180115-182626

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-xnqsh.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180116-101414/

**Comment by cockpituous** [comment]
Task aborted.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-0rt9w.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180116-125151/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180116-125151/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-h0jkf.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180116-134444/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180116-134444/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-b6lvv.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180116-142222/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180116-142222/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-17wnw.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-064040/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-064040/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-bjb38.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-071010/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-071010/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-0rt9w.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-110404/

**Comment by cockpituous** [comment]
image-refresh openshift done: https://github.com/cockpituous/cockpit/commits/image-refresh-openshift-20180117-113838

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-17wnw.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-122020/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-122020/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-73q6p.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-124242/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-124242/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-d09dd.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-140909/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-140909/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-d09dd.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-151212/

**Comment by mareklibra** [comment]
Recent findings:
- pre-pulled docker images are evaporating or at least loosing their tags
- in the past, this was happening due to lack of disk space inside the openshift VM image
- this time, the issue is unrelated to the actual disk size - disk size is 30G with plenty of freespace
- issue can be observed at `sync` command within the `zero-disk.setup` script
- suspection: when the backing loopdevice storing the docker images (`var/lib/docker/devicemapper/devicemapper`) runs out of space as a consequence of the `dd` command zero-ing all the remaining disk space, the images are being removed no matter there are is no additional `docker pull`

Any experience with that?
If I'm right, we might fix the issue by replacing the loopdevice by something else, more stable.

**Comment by mareklibra** [comment]
`sync` after `dd` and writing by `docker` might happen in parallel. So let's see in this commit

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-1dl5p.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180117-160202/

**Comment by cockpituous** [comment]
image-refresh openshift done: https://github.com/cockpituous/cockpit/commits/image-refresh-openshift-20180117-164343

**Comment by mareklibra** [comment]
Yes, the issue is caused by asynchronous write to the loopdevice competing with the `dd` filling the empty disk space. I will post a fix starting/stoping docker and syncing before `dd`.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-bjb38.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180118-074848/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180118-074848/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-h0jkf.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180118-083535/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180118-083535/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-bvw3m.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180118-120404/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180118-120404/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-j8nqh.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8414-20180119-075959/

**Comment by cockpituous** [comment]
image-refresh openshift done: https://github.com/cockpituous/cockpit/commits/image-refresh-openshift-20180119-083232

**Comment by mareklibra** [comment]
Image build succeeded.
Restarting tests with the new image: https://github.com/cockpituous/cockpit/commit/4b071e64790bed4bd0208875efe9442800797c0f

File: bots/images/scripts/lib/zero-disk.setup [hunk 1]
```diff
@@ -41,7 +41,6 @@ if [ -d "/var/cache" ]; then
 fi
 rm -rf /var/tmp/*
 rm -rf /var/log/journal/*
-
```
Comment (martinpitt) [inline bots/images/scripts/lib/zero-disk.setup]: Erratic whitespace removal (but fine to remove that  on merging).

File: bots/images/scripts/openshift.setup [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+        if eval "$@"; then
+            return 0
+        fi
+        sleep 6
+    done
+    exit 6
+}
+
+function docker_images_has() {
+    docker images | tr -s ' ' | cut -d ' ' --output-delimiter=: -f1,2 | grep -q "$1"
```
Comment (martinpitt) [inline bots/images/scripts/openshift.setup]: `docker images --format "{{.Repository}}:{{.Tag}}"` would be a bit more robust. You even already use that further down :-)

File: bots/images/scripts/openshift.setup [hunk 3]
```diff
@@ -278,8 +289,39 @@ printf 'AuthorizedKeysCommand /usr/local/bin/authorized-kube-keys --kubeconfig=/
...
+echo
+echo All images expected to be pulled
+cat /tmp/pulledImages
+echo
+
+# Verify all expected are actually pulled
+while read img ; do
+  echo Verify "$img"
+  grep "$img" /tmp/presentImages || (echo "Error: Image $img is missing" && exit 10)
+done < /tmp/pulledImages
```
Comment (martinpitt) [inline bots/images/scripts/openshift.setup]: This would be quite a bit simpler with using `join`.

**Comment by martinpitt** [comment]
debian-testing is green, just failed to tell github about it. So, yay!

**Pull request closed without merging**